test/harry/main/org/apache/cassandra/harry/checker/ModelChecker.java

```diff
@@ -212,20 +253,146 @@ public void map(ThrowingFunction<T, T> fn) throws Throwable
         }
     }
 
-    public static class State<MODEL, SUT>
+    public static class Pair<L, R>
     {
-        public final MODEL model;
-        public final SUT sut;
+        private static Pair<?, ?> UNCHANGED = new Pair<>(null, null);
+
+        public static <L, R> Pair<L, R> unchanged()
+        {
+            return (Pair<L, R>) UNCHANGED;
+        }
+
+        public final L l;
+        public final R r;
+
+        public Pair(L l, R r)
+        {
+            this.l = l;
+            this.r = r;
+        }
+
+        public Pair<L, R> next(L state)
+        {
+            return new Pair<>(state, this.r);
+        }
+    }
+
+    public class Simple
+    {
+        public Simple init(STATE state)
+        {
+            ModelChecker.this.init = new Pair<>(state, null);
+            return this;
+        }
+
+        public Simple beforeAll(ThrowingConsumer<STATE> beforeAll)
```

Comment:
nit: Some unused overloads in here, but you have a better idea of whether they'll be useful in the near future or not.

Reply:
Right; added `suppresswarnings` for now, I would like to believe the other overloads will be useful over time.